orttraining/orttraining/python/ort_trainer.py

```diff
@@ -696,6 +696,9 @@ def _init_session(self):
         if self.run_symbolic_shape_infer:
             self.onnx_model_ = SymbolicShapeInference.infer_shapes(self.onnx_model_, auto_merge=True, guess_output_rank=True)
 
+        # old ort session may already exists and occupies GPU memory when creating new session, this may cause OOM error.
+        # for example, load_state_dict will be called before returing the function, and it calls _init_session again
+        del self.session
```

Comment:
Move this to line 461, where `TrainingSession` is really instantiated

Reply:
good point.

Reply:
just realize that function "create_ort_training_session_with_optimizer" is not class member function, so this is the right place to delete self.session?